lewis/adapters/stream.py

```diff
@@ -163,8 +186,12 @@ def __init__(self, func, pattern, argument_mappings=None, return_mapping=None, d
             raise RuntimeError('Can not construct a Func-object from a non callable object.')
 
         self.func = func
-        self.raw_pattern = pattern
-        self.pattern = re.compile(b(pattern), 0) if pattern else None
+
+        self.raw_pattern = pattern.raw_pattern
+        self.pattern = re.compile(b(pattern.regex_pattern), 0)
```

Comment:
PS: I should also ask the IBEX team what the motivation behind `CmdBuilder` was... maybe if we had `scanf`, they wouldn't need it. Not in the office today, but I'll see if I can chat with them tomorrow... or maybe just drop them an email. But it would probably be easier to discuss in person.